### PR TITLE
Use .Release.Name in frontend.memcached.address to match service

### PR DESCRIPTION
**Pull request by elliottmb**
**What this PR does**: Fixes the mapping between the query-frontend and the memcached created for it. The other memcached instances spun up in this chart use `.Release.Name`.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: templates/_helpers.tpl [hunk 1]
```diff
@@ -109,7 +109,7 @@ Create configuration for frontend memcached configuration
 */}}
 {{- define "cortex.frontend-memcached" -}}
 {{- if index .Values "memcached-frontend" "enabled" }}
-- "-frontend.memcached.addresses=dns+{{ template "cortex.fullname" . }}-memcached-frontend.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:11211"
```
Comment (nschad) [inline templates/_helpers.tpl:112]: I can see why you did this but I actually think this is the correct way and the usage of .Release.Name is wrong. So I rather have the other memcache url's corrected.

**But** I'll verify that tomorrow.

Comment (nschad) [reply templates/_helpers.tpl:112]: Yep if you were to use for examle `fullNameOverride` it corrently wouldn't work with `.Release.Name`. We need to change everything to `{{ template "cortex.fullname" . }}` @elliottmb

**Review by nschad — CHANGES_REQUESTED** [review]
see comment above

**Comment by elliottmb** [comment]
@nschad Totally open to that change. This PR was to fix a bug we ran into where the query-frontend doesn't properly start if you try to enable this due to the mismatch of what the service is and what this was setting, so was wanting to fix that without changing the world. 

Would you want me to edit this PR to make that larger change or could that potentially be a follow up PR with this one still merged as an intermediate bug fix?

**Comment by nschad** [comment]
> Would you be up to merging this PR as an intermediate bug fix given it's broken presently? I can follow up this PR to make that larger change at a later date.

I  don't feel comfortable doing this. What are your thoughts on this @kd7lxl ?

**Comment by nschad** [comment]
> For context, this is the current behavior without the change.
> 
> ```
> level=warn ts=2022-03-16T15:41:16.49671671Z caller=resolver.go:133 msg="IP address lookup yielded no results. No host found or no addresses found" host=v1-cortex-memcached-frontend.cortex.svc.cluster.local
> level=error ts=2022-03-16T15:42:16.49494025Z caller=resolver.go:86 msg="failed to lookup IP addresses" host=v1-cortex-memcached-frontend.cortex.svc.cluster.local err="lookup v1-cortex-memcached-frontend.cortex.svc.cluster.local on 172.26.168.10:53: no such host"
> level=warn ts=2022-03-16T15:42:16.495017428Z caller=resolver.go:133 msg="IP address lookup yielded no results. No host found or no addresses found" host=v1-cortex-memcached-frontend.cortex.svc.cluster.local
> level=error ts=2022-03-16T15:43:16.502803721Z caller=resolver.go:86 msg="failed to lookup IP addresses" host=v1-cortex-memcached-frontend.cortex.svc.cluster.local err="lookup v1-cortex-memcached-frontend.cortex.svc.cluster.local on 172.26.168.10:53: no such host"
> ```
> 
> because
> 
> ```
> $ kubectl -n cortex get service
> NAME                                   TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
> v1-cortex-compactor                    ClusterIP   172.26.169.219   <none>        8080/TCP            8d
> ...
> v1-cortex-store-gateway-headless       ClusterIP   None             <none>        9095/TCP            8d
> v1-memcached-blocks                    ClusterIP   None             <none>        11211/TCP           8d
> v1-memcached-blocks-index              ClusterIP   None             <none>        11211/TCP           8d
> v1-memcached-blocks-index-metrics      ClusterIP   172.26.169.144   <none>        9150/TCP            8d
> v1-memcached-blocks-metadata           ClusterIP   None             <none>        11211/TCP           8d
> v1-memcached-blocks-metadata-metrics   ClusterIP   172.26.171.126   <none>        9150/TCP            8d
> v1-memcached-blocks-metrics            ClusterIP   172.26.170.195   <none>        9150/TCP            8d
> v1-memcached-frontend                  ClusterIP   None             <none>        11211/TCP           45h
> v1-memcached-frontend-metrics          ClusterIP   172.26.171.108   <none>        9150/TCP            45h
> ```

Can you tell me the name of your helm-release?

**Comment by elliottmb** [comment]
I believe it's simply `v1`.

**Comment by nschad** [comment]
@elliottmb Okay I think this pr would fix the problem. My thoughts appear to be wrong

**Review by nschad — APPROVED** [review]
thank you. Sorry for the confusion

**Review by kd7lxl — APPROVED** [review]

**Pull request merged**